lrpar/examples/calc_wasm/src/lib.rs

```diff
@@ -0,0 +1,117 @@
+#[cfg(all(
+    target_arch = "wasm32",
+    target_os = "unknown",
+    target_vendor = "unknown"
+))]
+use wasm_bindgen::prelude::*;
+
+use lrlex::lrlex_mod;
+use lrpar::lrpar_mod;
+
+// Using `lrlex_mod!` brings the lexer for `calc.l` into scope. By default the module name will be
+// `calc_l` (i.e. the file name, minus any extensions, with a suffix of `_l`).
+lrlex_mod!("calc.l");
+// Using `lrpar_mod!` brings the parser for `calc.y` into scope. By default the module name will be
+// `calc_y` (i.e. the file name, minus any extensions, with a suffix of `_y`).
+lrpar_mod!("calc.y");
+
+#[cfg_attr(
+    all(
+        target_arch = "wasm32",
+        target_os = "unknown",
+        target_vendor = "unknown"
+    ),
+    wasm_bindgen
+)]
+pub fn calculate(l: &str) -> Result<u64, String> {
```

Comment:
A big reason this is as ugly as it is is that on `wasm32-unknown-unknown` the `Err` variant must be able to be converted into a JsonValue, hence the `String` error.

Uncertain about `wasm32-wasip2`.